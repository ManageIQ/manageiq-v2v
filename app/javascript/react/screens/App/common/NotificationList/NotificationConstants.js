const V2V_NOTIFICATION_ADD = 'V2V_NOTIFICATION_ADD';
const V2V_NOTIFICATION_REMOVE = 'V2V_NOTIFICATION_REMOVE';

export { V2V_NOTIFICATION_ADD, V2V_NOTIFICATION_REMOVE };
