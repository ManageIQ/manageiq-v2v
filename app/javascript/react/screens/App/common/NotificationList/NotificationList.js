import React from 'react';
import PropTypes from 'prop-types';
import {
  TimedToastNotification,
  ToastNotificationList
} from 'patternfly-react';
import { Link } from 'react-router-dom';

class NotificationList extends React.Component {
  render() {
    const { notifications, removeNotificationAction } = this.props;

    return (
      <ToastNotificationList>
        {notifications &&
          notifications.map((notification, index) => (
            <TimedToastNotification
              key={index}
              type={notification.notificationType}
              persistent={notification.persistent}
              timerdelay={notification.timerdelay}
              onDismiss={() => removeNotificationAction({ notification })}
              onMouseLeave={() => removeNotificationAction({ notification })}
            >
              <span>
                {notification.header && <strong>{notification.header}</strong>}
                {notification.message}
              </span>
              {notification.actionEnabled && (
                <div className="pull-right toast-pf-action">
                  <Link to={`/migration/plan/${notification.data.id}`}>
                    {__('View Details')}
                  </Link>
                </div>
              )}
            </TimedToastNotification>
          ))}
      </ToastNotificationList>
    );
  }
}

NotificationList.propTypes = {
  removeNotificationAction: PropTypes.func,
  notifications: PropTypes.array
};

NotificationList.defaultProps = {
  notifications: []
};

export default NotificationList;
