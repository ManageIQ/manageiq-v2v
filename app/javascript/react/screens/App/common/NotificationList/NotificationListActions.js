import {
  V2V_NOTIFICATION_ADD,
  V2V_NOTIFICATION_REMOVE
} from './NotificationConstants';

export const addNotificationAction = notification => dispatch => {
  dispatch({
    type: V2V_NOTIFICATION_ADD,
    header: notification.header,
    message: notification.message,
    notificationType: notification.notificationType,
    data: notification.data,
    persistent: notification.persistent,
    timerdelay: notification.timerdelay,
    actionEnabled: notification.actionEnabled
  });
};

export const removeNotificationAction = key => dispatch => {
  dispatch({
    type: V2V_NOTIFICATION_REMOVE,
    key
  });
};
