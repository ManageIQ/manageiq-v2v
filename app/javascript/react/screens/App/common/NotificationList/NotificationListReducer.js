import Immutable from 'seamless-immutable';

import {
  V2V_NOTIFICATION_ADD,
  V2V_NOTIFICATION_REMOVE
} from './NotificationConstants';

const initialState = Immutable({
  notifications: []
});

export default (state = initialState, action) => {
  const newNotification = {
    header: action.header,
    message: action.message,
    notificationType: action.notificationType,
    data: action.data,
    persistent: action.persistent,
    timerdelay: action.timerdelay,
    actionEnabled: action.actionEnabled
  };

  switch (action.type) {
    case V2V_NOTIFICATION_ADD:
      return state.set('notifications', [
        ...Immutable.asMutable(state.notifications),
        newNotification
      ]);
    case V2V_NOTIFICATION_REMOVE:
      return state.set(
        'notifications',
        Immutable.asMutable(state.notifications).filter(
          notification =>
            action.key.notification.message !== notification.message
        )
      );
    default:
      return state;
  }
};
