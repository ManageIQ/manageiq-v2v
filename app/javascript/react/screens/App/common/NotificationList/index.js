import { connect } from 'react-redux';
import NotificationList from './NotificationList';
import * as NotificationListActions from './NotificationListActions';
import reducer from './NotificationListReducer';

export const reducers = { notifications: reducer };

const mapStateToProps = ({ notifications }, ownProps) => ({
  ...notifications,
  ...ownProps.data
});

const mergeProps = (stateProps, dispatchProps, ownProps) =>
  Object.assign(stateProps, ownProps.data, dispatchProps);

export default connect(mapStateToProps, NotificationListActions, mergeProps)(
  NotificationList
);
