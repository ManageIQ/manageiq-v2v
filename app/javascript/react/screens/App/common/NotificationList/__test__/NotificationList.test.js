import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import NotificationList from '../NotificationList';
import NotificationListReducer from '../NotificationListReducer';

describe('NotificationList Component', () => {
  test('correctly empty renders', () => {
    const wrapper = mount(<NotificationList />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('NotificationsListReducer initial state', () => {
  it('should return the initial state', () => {
    const initialState = {
      notifications: []
    };

    expect(NotificationListReducer(undefined, {})).toEqual(initialState);
  });
});
