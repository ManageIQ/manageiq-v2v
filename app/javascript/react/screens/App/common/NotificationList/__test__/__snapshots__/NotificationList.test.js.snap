// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationList Component correctly empty renders 1`] = `
<NotificationList
  notifications={Array []}
>
  <ToastNotificationList
    className=""
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="toast-notifications-list-pf"
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    />
  </ToastNotificationList>
</NotificationList>
`;
